src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/GetLegendGraphicTest.java

```diff
@@ -78,6 +95,42 @@ public void testEnv() throws Exception {
     }
     
     /**
+     * Tests an custom legend graphic
+     */
+    @Test
+    public void testCustomLegend() throws Exception {
+        String base = "wms?service=WMS&version=1.1.1&request=GetLegendGraphic" +
+                        "&layer=sf:states&style=custom" +
+                        "&format=image/png&width=22&height=22";
+        
+        BufferedImage image = getAsImage(base, "image/png");        
+        Resource resource = getResourceLoader().get("styles/legend.png");
+        BufferedImage expected = ImageIO.read( resource.file() );
+        
+        assertEquals( getPixelColor(expected,10,2).getRGB(), getPixelColor(image,10,2).getRGB() );
+        
+        // test rescale
+        base = "wms?service=WMS&version=1.1.1&request=GetLegendGraphic" +
+                "&layer=sf:states&style=custom" +
+                "&format=image/png&width=16&height=16";
+
+        image = getAsImage(base, "image/png");        
+        
+        String user_home = System.getProperty("user.home");
+        File home = new File(user_home);
+        ImageIO.write(expected, "png",  new File(home,"expected.png"));
```

Comment:
@jodygarnett @tbarsballe please do not write test images to user home directory. Fixed in 438097734641e2d99a2346d186179b0bd0338e08 on master. These images were not used in the test and I assume that they were for test development and inadvertently included in this PR. At the least, ensure that test images are confined to module target so they can be cleaned. See gt-render for examples of how to write test fixtures that enable interactive development of tests and by default write no files.